mobile/src/views/MyICardView.jsx

```diff
@@ -1,34 +1,100 @@
-import React, {useContext} from 'react';
-
+import React, {useContext, useEffect, useState, } from 'react';
+import { View,  RefreshControl, StyleSheet,  ScrollView, ActivityIndicator } from 'react-native';
 import AuthContext from '../context/AuthContext';
 import MyICardPage from '../components/shared/ICardPage';
 import VerificationView from './VerificationView'
 import {NavigationContainer} from '@react-navigation/native';
 import {createNativeStackNavigator} from '@react-navigation/native-stack';
+//import * as StudentFunctions from '../../../backend/src/controllers/Students';
+
 
-let finalStatus = 'inactive';
+let finalStatus;
+let message;
 
 const Stack = createNativeStackNavigator();
 
+import {API_ROUTE, API_KEY} from '@env';
+import { storeUser } from '../utilites/StoreUser';
+
+const authRoute = API_ROUTE;
+const apiKey = API_KEY;
+
 const MyICard = ({navigation}) => {
-  const {user, _} = useContext(AuthContext);
+  const {user, setUser} = useContext(AuthContext);
+  const [checkStatus, setCheckStatus] = useState(true);
+  const [refreshing, setRefreshing] = useState(false);
+
+  const loadUserData = async () => {
+
+    setRefreshing(true);
+    fetch(authRoute + `api/students/${user.id}`, {
+      method: 'GET',
+      headers: {'x-api-key': apiKey, 'jwt-token': user.key},
+    })
+    .then(result => {
+      return result.json();
+    })
+    .then(data => {
+      data["key"] = user.key;
+      data["id"] = data["_id"];
+      data["verification_image"] == ""  || data["verification_image"] == undefined ? data["verification_image"] = "" : null;
+      delete data["_id"];
+      setRefreshing(false);
+      setUser(data);
+      storeUser(data);
+      setCheckStatus(true);
+    })
+    .catch((error) => {
+      console.error(error);
+    });
+}
 
-  if (user == null) return <></>;
-  if (user.verify) finalStatus = 'inactive';
-  if (user.isaf_status) finalStatus = 'active';
+if (user == null) return <></>;
 
-  if (!user.isaf_status && !user.verify && user.verification_image)
+
+const statusCheck = () => {
+  if(user.isaf_status == true && user.verify_status == true){
+    finalStatus = 'active';
+  }else if(user.isaf_status == false && user.verify_status == true){
+    finalStatus = 'inactive';
+    message = 'Something went wrong. Try to re-submit the screenshot image or contact ISA at "isa.general@ualberta.ca"';
+  }else if(user.isaf_status == false && user.verify_status == false && user.verification_image == ""){
+    finalStatus = 'inactive';
+    message = 'Your account is unverified. Please go through the verification process';
+  }
+  else if(user.isaf_status == false && user.verify_status == false && user.verification_image != ""){
```

Comment:
I think this logic is fine but if these attributes are booleans we can just use them directly instead of comparing to `== true` and `==false`

Reply:
Additionally I think its fine since the message doesn't change after render, but if you find it will, we should change it to a useState property instead.

Reply:
There were similar if checks (smaller ones obviously) whose template I just copied to make these if conditions. I tried useState but the logic got a bit more confusing or more complicated than required. Obviously, given a little bit more time, it would have been easy to solve but I was like "if this way exists that seems much easier to implement, why not just do this instead". So I stuck with this. I've checked it in every possible way and it works fine. Should I just change the formatting of the if conditions?

Reply:
by formatting I mean the first thing you mentioned about the ==true and ==false

Reply:
Yea let's ignore the useState thing for now, but we should probably change the == true, == false it's unnecessary.